src/metadataGeneration/metadataGenerator.ts

```diff
@@ -56,6 +56,34 @@ export class MetadataGenerator {
       }
 
       ts.forEachChild(sf, node => {
+        /**
+         * If we declare a namespace within a module, like we do in `tsoaTestModule.d.ts`,
+         * we need to explicitly get the children of the module declaration
+         * (`declare module 'tsoaTest'`) - which are the moduleBlock statements,
+         * because otherwise our type resolver cannot iterate over namespaces defined in that module.
+         */
+        if (ts.isModuleDeclaration(node)) {
+          const moduleDeclaration = node as ts.ModuleDeclaration;
```

Comment:
TS should figure this out without explicit type casting

Reply:
Fixed.